src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/JsonHubProtocol.cs

```diff
@@ -377,7 +377,16 @@ private object[] BindArguments(JArray args, IReadOnlyList<Type> paramTypes)
 
         internal static JsonSerializerSettings CreateDefaultSerializerSettings()
         {
-            return new JsonSerializerSettings { ContractResolver = new CamelCasePropertyNamesContractResolver() };
+            return new JsonSerializerSettings
+            {
+                ContractResolver = new DefaultContractResolver()
+                {
+                    NamingStrategy = new CamelCaseNamingStrategy()
+                    {
+                        ProcessDictionaryKeys = true
```

Comment:
Does MVC do this?

Reply:
Probably not, but MVC doesn't have the intermediate bus in the way. I'm not sure this is the best way to solve this yet, but we need to do something to resolve the underlying issue :)

Reply:
No, MVC doesn't. People complain that it doesn't, but people would also complain if it did.